.github/workflows/ci.yml

```diff
@@ -265,17 +265,23 @@ jobs:
             name=${{ steps.prepare-meta.outputs.image }}
           labels: |
             ${{ fromJSON(steps.prepare-meta.outputs.labels) }}
+
       - name: Build Docker image
         id: build-docker-image
         if: steps.prepare-meta.outputs.has-image
         timeout-minutes: 120
-        run: cargo xtask build-docker-image -v "${TARGET}${SUB:+.$SUB}"
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 120
+          max_attempts: 3
+          retry_on: error
+          command: cargo xtask build-docker-image -v "${TARGET}${SUB:+.$SUB}"
```

Comment:
I'm not 100% sure I like this, what if it fails for a valid reason? Then we'll retry unconditionally, and there'll be no cache (I think) for the failed steps so it'll triple up on the errors.